src/components/editor/PageUpdater.js

```diff
@@ -22,14 +22,23 @@ export default class PageUpdater {
     this.data = resp.data;
   }
 
+  async reRequestDocument() {
+    throw Error('Not implemented');
+  }
+
   async updateDataProperty(path, property, newValue) {
+    // Note that this only changes
```

Comment:
The more I think about it, the more I'd be ok with lodash to make this easier.

A code comment about this function would not be terrible.